lib/postzord/receiver/local_batch.rb

```diff
@@ -25,6 +25,15 @@ def receive!
     # 09/27/11 this is slow
     notify_users
 
+    if @object.respond_to?(:type) && @object.type == "Reshare"
```

Comment:
We have the constructed object here, no? Why not duck type on `root` like below?